File: AOJ/0419/molting.py
L, N = map(int, input().split())
snake = input()
oocnt = 0
for i in range(len(snake) - 1):
    if snake[i] == 'o' and snake[i + 1] == 'o' :
        oocnt = oocnt + 1
total = 0
for i in range(N):
    total = total + oocnt
    oocnt = oocnt * 2
print(3 * total + L)

File: AOJ/0419/molting.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
  int L, N;
  string snake;
  cin >> L >> N >> snake;

  long long oocnt = 0;
  for ( int i = 0; i < L - 1; i++ ) {
    if ( snake.substr(i, 2) == "oo" )
      ++oocnt;
  }

  long long total = 0;
  for ( int i = 0; i < N; i++ ) {
    total += oocnt;
    oocnt *= 2;
  }

  cout << 3 * total + L << endl;
  return 0;
}
